theano/gpuarray/dnn.py

```diff
@@ -1626,11 +1630,12 @@ def apply(self, fgraph):
 
 @register_opt('cudnn', 'fast_compile')
 @op_lifter([SoftmaxGrad])
-def local_softmax_dnn_grad(node, ctx_name):
+#@register_opt2([SoftmaxGrad], 'fast_compile')
+def local_softmax_dnn_grad(op, ctx_name, inputs):
```

Comment:
Add the tag 'cudnn', as done in the register_opt. In fact, I think we should add the same tag in both register_opt and register_opt2 for all optimization. This will prevent this type of problems